napari/benchmarks/benchmark_shapes_layer.py

```diff
@@ -27,48 +27,164 @@
     from napari.benchmarks.utils import Skip
 
 
-class Shapes2DSuite:
+class BackendType(Enum):
+    bermuda = auto()  # data processing using bermuda, (https://github.com/napari/bermuda), preprocessing using numba
+    partsegcore = auto()  # data preprocessing using PartSegCore (https://partseg.github.io), edge triangulation using numba
+    triangle = auto()  # data preprocessing using numba, edge triangulation using numba, triangulation using triangle
+    numba = auto()  # data preprocessing and edge triangulation using numba, triangulation using vispy
+    pure_python = auto()  # data preprocessing, edge triangulation, and triangulation using python and vispy
+
+    def __str__(self):
+        return self.name
+
+    def __repr__(self):
+        return self.name
+
+
+backend_list = list(BackendType)
+backend_list.remove(
+    BackendType.bermuda
+)  # remove when merge bermuda backend PR #7747
+
+backend_list_complex = [BackendType.partsegcore, BackendType.numba]
+# pure python backend is too slow for large datasets
+# triangle backend requires deduplication of vertices in #6654
+
+
+class _BackendSelection:
+    triangulate: (
+        Callable | None
+    )  # store the original triangulate function, to be restored
+    prev_settings: bool  # store the state of NapariSettings.experimental.compiled_triangulation, to be restored
+    prev_numba: dict[
+        str, Callable
+    ]  # Dict form function name to function. It is used to store original numba functions from _accelerated_triangulate_dispatch
+
+    def _disable_numba(self):
+        """Replace numba jit functions with pure python functions."""
+        try:
+            from napari.layers.shapes import (
+                _accelerated_triangulate_dispatch as _triangle_dispatch,
+            )
+        except ImportError:
+            return
+
+        for name in _triangle_dispatch.__all__:
+            # This part of the code assumes that all pure python functions
+            # from _accelerated_triangulate_dispatch that are used when import from
+            # _accelerated_triangulate module fails because of missing numba
+            # are named as the function that they replace with added _py suffix.
+            # So we could write generic code to replace all of them.
```

Comment:
```suggestion
            # This code assumes that a pure python function name appends a _py suffix
            # to the numba jit function name. This is helpful if numba is missing, and it is
            # necessary to fall back to pure python functions.
```

Reply:
No. 
In this comment I point to these lines:

https://github.com/napari/napari/blob/4c9242de028ed18305fc2dae2db32745694f6059/napari/layers/shapes/_accelerated_triangulate_dispatch.py#L299-L302

I have fixed line 301 suffix 

You may see that the exported name differs only with suffix `_py`, so it allows to replace numba version with python version by monkeypatching.